src/errors.rs

```diff
@@ -8,6 +8,8 @@
 
 use dusk_bytes::Error as DuskBytesError;
 use dusk_jubjub::JubJubScalar;
+use dusk_plonk::error::Error as PlonkError;
+use dusk_poseidon::Error as PoseidonError;
 #[cfg(feature = "std")]
 use std::fmt;
```

Comment:
Are you planning to have a different PR to refactor the error removing `std` specific flags? Since you're refactoring the error module here, I was assuming you would handle that too. Or there is a reason why you're not using `core::fmt`?

Reply:
Exactly! The last issue is to re-work the `std`/`alloc` feature-set